AzFileDiagnostics/Linux/AzFileDiagnostics.sh

```diff
@@ -0,0 +1,746 @@
+ver_lt() { test "$(printf  "$1\n$2"  | sort -V | head -n 1)" != "$2"; }
+
+
+LOGDIR="MSFileMountDiagLog"
+if [ ! -d "$LOGDIR" ]; then
+	mkdir "$LOGDIR"
+fi
+
+print_log "Create a folder MSFileMountDiagLog to save the script output"
+
+
+## Verify Linux distribution version. There is a list of recommended images for use
+DISTNAME=''
+DISTVER=''
+KERVER=''
+
+command -v lsb_release >/dev/null 2>&1  && DISTNAME=$(lsb_release -i | cut -d : -f 2) || DISTNAME=$(cat /etc/*release | grep \\bNAME= | cut -d = -f 2)
+command -v lsb_release >/dev/null 2>&1  && DISTVER=$(lsb_release -r |  grep -o  [0-9\\.]\\+ )  || DISTVER=$(cat /etc/*release | grep \\bVERSION_ID= | grep -o  [0-9\\.]\\+)
+
+#DISTNAME=$(cat /etc/*release | grep \\bNAME= | cut -d = -f 2)
+#DISTVER=$(cat /etc/*release | grep \\bVERSION_ID= | grep -o  [0-9\\.]\\+)
+KERVER=$(uname -r | cut -d - -f 1)
+KERVEREXTRA=$(uname -r)
+
+#Trim the whitespace
+DISTNAME=$(echo $DISTNAME)
+DISTVER=$(echo $DISTVER)
+KERVER=$(echo $KERVER)
+KERVEREXTRA=$(echo $KERVEREXTRA)
+
+print_log "Client running with $DISTNAME version $DISTVER, kernel version is $KERVEREXTRA"
+
+case $DISTNAME  in
+	*Redhat* )
+		if ( ver_lt $DISTVER '8' ); then
+			print_log "We recommend running following Linux Distributions: Ubuntu Server 18.04+ | RHEL 8+ | CentOS 7.6+ | Debian 9 | openSUSE 15+ | SUSE Linux Enterprise Server 15, please refer to https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux for more information" "warning"
+		fi
+		#kernle version check
+		if ( ! ver_lt $DISTVER '8' ); then
+			if ( ver_lt $KERVER '4.18.0'); then
+				print_log "For RHEL 8, we recommend running Kernel with version 4.18.0+" "warning"
+			fi
+		fi
+		;;
+
+	*CentOS* )
+		if ( ver_lt $DISTVER '7.6' ); then
+			print_log "We recommend running following Linux Distributions: Ubuntu Server 18.04+ | RHEL 8+ | CentOS 7.6+ | Debian 9 | openSUSE 15+ | SUSE Linux Enterprise Server 15, please refer to https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux for more information" "warning"
+		fi
+
+		#kernle version check
+		if ( ver_gt $DISTVER '7.5' ); then
+			if ( ver_lt $DISTVER '7.7' ); then
+				if ( ver_lt $KERVER '3.10.0'); then
+					print_log "For CentOS 7.6, we recommend running Kernel with version 3.10.0+" "warning"
+				fi
+			fi
+		fi
+		;;
+
+	*Ubuntu* )
+		echo "Current Ubuntu distribution version is:  $DISTVER"
+		if ( ver_lt $DISTVER '18.04' ); then
+			print_log "We recommend running following Linux Distributions: Ubuntu Server 18.04+ | RHEL 8+ | CentOS 7.6+ | Debian 9 | openSUSE 15+ | SUSE Linux Enterprise Server 15, please refer to https://docs.microsoft.com/en-us/azure/storage/files/storage-how-to-use-files-linux for more information" "warning"
+		fi
+
+		# kernle version check for ubuntu 18.04
+		if ( ver_gt $DISTVER '18.03' ); then
+			if ( ver_lt $DISTVER '18.05' ); then
+				echo "Current Kernel version is: $KERVER"
+				if ( ver_lt $KERVER '4.18.0'); then
```

Comment:
5.0 instead of 4.18 for Ubuntu and also typo: kernle -> kernel couple lines above